### ISSUE TITLE
Things to add to the style guide

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#1018]
Some things that I keep meaning to add to the style-guide but never get around to:
- layout of `data` declarations
- layout of `record` declarations
- avoid `.` patterns for inferred arguments now that they're optional in Agda
- align function arguments
- add empty implicit brackets `{_}` in other cases if the implicit is ever used

**Comment by pnlph** [comment]
I could work on this, any hint on how to start?

**Comment by laMudri** [comment]
Is the style guide also the right place for explaining `Core` vs `Base` vs `Properties` vs plain; `Structures`, `Bundles`, &c; and other file placement conventions?

**Comment by MatthewDaggitt** [comment]
> Is the style guide also the right place for explaining Core vs Base vs Properties vs plain; Structures, Bundles, &c; and other file placement conventions?

I don't think so. That should be in the `README` folder. I've got a branch explaining all this that I'll try and push in the next few days.

**Comment by MatthewDaggitt** [comment]
Flushed most of the list in #1110. If people don't object, I'm going to keep this issue open as a handy and quick place to add items as they occur to me.

**Comment by jamesmckinna** [comment]
And if there are 'new' items (nesting depth for `module`s, ...?), then the currently-open PR #2270 is open to additional commits...

**Comment by jamesmckinna** [comment]
Pattern synonyms in `Base` modules: #2262
